mentha2pdb/mentha2pdb.py

```
@@ -844,29 +831,17 @@ def copy_folder(ex, id1, id2, af_folder_path):
             f'source folder does not exist \n {from_path}'
         print(s)
 
-
-def rename_folders_based_on_uniprot(ensg_to_uniprot_dict, base_path='AF_Huri_HuMAP'):
-    #Rename folders from ENSG to UniProt IDs .
-
+def rename_pair_folder_direct(ensg1, ensg2, up1, up2, base_path='AF_Huri_HuMAP'):
     huri_path = Path(base_path, 'Huri_dimers')
     humap_path = Path(base_path, 'HuMAP_dimers')
 
     for sub_path in [huri_path, humap_path]:
-        if sub_path.exists():
-            for folder in sub_path.iterdir():
-                if folder.is_dir():
-                    parts = folder.name.split('-')
-                    if len(parts) == 2:
-                        ensg1, ensg2 = parts
-                        uniprot1 = ensg_to_uniprot_dict.get(ensg1)
-                        uniprot2 = ensg_to_uniprot_dict.get(ensg2)
-                        if uniprot1 and uniprot2:
-                            new_name = f"{uniprot1}-{uniprot2}"
-                            new_folder_path = folder.parent / new_name
-                            folder.rename(new_folder_path)
-                            print(f"Renamed {folder.name} to {new_name}")
-
-
+        old_folder = sub_path / f"{ensg1}-{ensg2}"
+        if old_folder.exists():
+            new_folder = sub_path / f"{up1}-{up2}"
+            if not new_folder.exists():
+                old_folder.rename(new_folder)
+                print(f"Renamed {old_folder} → {new_folder}")
```

is it ok to do nothing if the old folder doesn't exist or the new folder already exists? Can't this be a sign that we are doing something problematic? I'm not sure it's ok that these conditions fail silently

---

You are right it is not ok. Only in cases of humap folders we expect them to be already in the up1-up2 format, but this function is only called for Huri folders that need renaming.